### DISCUSSION TITLE
Finding deploy hook as author from environment variable?

### DISCUSSION THREAD

**Discussion by jessetan** [vercel/community#2038] [Help] [answered]
### Summary

When Vercel creates a deployment, the overview shows the author of the commit that triggered the build.
For deployments started from a deploy webhook, it shows "via Deploy Hook".
<img width="275" alt="Screenshot 2023-04-11 at 12 26 09" src="https://user-images.githubusercontent.com/3284111/231133130-19810037-c1d5-4b99-8186-058b9ca25fd6.png">

Is there [any environment variable](https://vercel.com/docs/concepts/projects/environment-variables/system-environment-variables) that exposes this info and tells me that a build was triggered by a webhook?

**Comment by amyegan** [comment, marked as answer]
Hi @jessetan! There is not currently an environment variable that exposes whether the build was triggered by a Deploy Hook

**Reply by Tobi-mmt** [reply]
It would be super nice if there is a way to find out if the deployment was triggered by the hook or not.
In my case, for example:
I have a monorepo and want to skip any deployment if the code of the subfolder did not change. So I activated the "skip deploy step" option.
So someone passes a code change to the mono repo only the changed apps are deployed.

But if the CMS now triggers a deployment the build step is also skipped because the code did not change only the content. 
For me it would be super helpful if I can implement a logic in the "skip deploy step" field that if the deployment is triggered by a hook, it should be deployed in every case, even if no code changed.

**Reply by amyegan** [reply]
While there isn't currently an environment variable dedicated to the deployment trigger, you can bypass the "Ignored Build Step" script if you use the CLI. You can kick off the deployment from your local environment using [`vercel deploy --prod`](https://vercel.com/docs/cli/deploy#prod) or from GitHub Actions with a [custom workflow](https://vercel.com/guides/using-vercel-cli-for-custom-workflows)

**Reply by Tobi-mmt** [reply]
@amyegan thank you for your reply. I see that I can byass the ignore step by deploying it manually with the CLI. But I can not see how this can resolve my problem with the deploy hook.
As far as I know [there is no trigger for github actions](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) from outside github. So I can not tell the CMS to trigger a github action.
I don't know how to use your information to let only the CMS skip the "ignore build step" within an automated pipeline.

**Reply by amyegan** [reply]
If you're using an automated pipeline, you may be able to make use of the CLI with whichever automation tool you are using. With GitHub actions, you can use the [`workflow_dispatch`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch) trigger to trigger a workflow run using the GitHub API. 

For content changes, [Incremental Static Regeneration](https://vercel.com/docs/concepts/incremental-static-regeneration/overview#) is a good option. Revalidation can happen automatically in the background over time, or on-demand revalidation can be triggered via URL to update displayed content. I've also seen people use [Cron Jobs](https://vercel.com/docs/cron-jobs) or a GitHub workflow with a [`schedule`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule) trigger to revalidate daily.

**Reply by amyegan** [reply]
Another discussion just reminded me that `?buildCache=false` might work for this situation. You can find more info on that here: https://vercel.com/docs/concepts/deployments/deploy-hooks#build-cache

**Reply by nfm** [reply]
I just ran into this and have the exact same use case as @Tobi-mmt - we have a monorepo, and only want to trigger a build if a subdirectory has changed, or via a deploy hook from our headless CMS.

Haven't got a solution for this yet. If Vercel exposed an env var when a build was triggered from a deploy hook I could change the 'Ignored build step' config to handle this correctly.
